lib/mongo/protocol/msg.rb

```diff
@@ -298,6 +298,17 @@ def maybe_add_server_api(server_api)
         Msg.new(@flags, @options, main_document, *@sequences)
       end
 
+      def number_returned
+        if doc = documents.first
+          if cursor = doc['cursor']
+            if batch = cursor['firstBatch'] || cursor['nextBatch']
+              return batch.length
+            end
+          end
+        end
+        0
```

Comment:
What would happen if this line instead raised `NotImplementedError`?

Reply:
I just started I patch with this, I'll let you know what the results are: [Evergreen Patch](https://spruce.mongodb.com/version/61ae43250305b925053f2992/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

Reply:
There are a lot of failures at this test:
```
1) Mongo::Collection::View::MapReduce#each when the view has a limit applies the limit to the map/reduce
     Failure/Error: raise NotImplementedError
     
     NotImplementedError:
       NotImplementedError
     # ./lib/mongo/protocol/msg.rb:309:in `number_returned'
     # ./lib/mongo/operation/result.rb:252:in `returned_count'
     # ./lib/mongo/cursor.rb:439:in `process'
     # ./lib/mongo/cursor.rb:206:in `try_next'
     # ./lib/mongo/cursor.rb:169:in `block in each'
     # ./lib/mongo/cursor.rb:168:in `loop'
     # ./lib/mongo/cursor.rb:168:in `each'
     # ./lib/mongo/collection/view/map_reduce.rb:79:in `each'
     # ./spec/mongo/collection/view/map_reduce_spec.rb:427:in `block (4 levels) in <top (required)>'
     # ./spec/lite_spec_helper.rb:131:in `block (3 levels) in <top (required)>'
     # ./spec/lite_spec_helper.rb:130:in `block (2 levels) in <top (required)>'
     # ./spec/lite_spec_helper.rb:113:in `block (2 levels) in <top (required)>'
     # ./spec/support/background_thread_registry.rb:65:in `block (2 levels) in <top (required)>'
```